capsules/src/i2c_master_slave_driver.rs

```diff
@@ -311,26 +299,26 @@ impl Driver for I2CMasterSlaveDriver<'_> {
                 let len = (data >> 16) & 0xFFFF;
 
                 self.app.map(|app| {
-                    app.master_rx_buffer.as_mut().map(|app_rx| {
+                    app.master_rx_buffer.map_or(0, |app_rx| {
```

Comment:
```suggestion
                    app.master_rx_buffer.map(|app_rx| {
```

Reply:
There is no `map` -- did you try compiling this?

Reply:
Like the `mut_map`'s above, this wasn't intended to apply directly (yet?), but to flag instances where our non-handling of errors has regressed. Ideally probably, all of these would be early-exit with error, but adding that much new error handling might be beyond the scope of 2.0 (although if we don't fix now, will we ever?). If we're not going to add error-handling, I think we should add the needed `map` methods so there isn't a question of what the meaningless `0` is in the future.

Reply:
Disagree. The problem is that a `map` allows implementations to (usually incorrectly) ignore error cases. I'm as guilty of this as anyone else. I think we want the API to force people to handle errors. If you see something weird like this, it's because the code is ignoring the error case.

We could just add a comment saying that this error case isn't being handled and the 0 should be handled better.

Reply:
I went through this capsule. The basic problem is that it has logic to perform transfers differently sized than what was requested, but does not report when this happens. E.g., it is possible to request a read of 20 and only read 8 bytes; 8 is never returned to user space. This happens, for example, if you pass a buffer of length 8.

Fixing this requires changing the system call API for this capsule.